File: frontend/src/types/registration.ts
import type { EventListItem } from './event';

export type RegistrationStatus = 'CONFIRMED' | 'CANCELLED';

export interface RegistrationRead {
  id: number;
  user_id: number;
  event_id: number;
  status: RegistrationStatus;
  created_at: string;
  updated_at: string;
}

export interface RegistrationWithEvent extends RegistrationRead {
  event: EventListItem;
}

export interface RegistrationsPage {
  items: RegistrationWithEvent[];
  total: number;
  limit: number;
  offset: number;
}

File: backend/app/schemas/registration.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.registration import RegistrationStatus
from app.schemas.event import EventListItem


class RegistrationRead(BaseModel):
    id: int
    user_id: int
    event_id: int
    status: RegistrationStatus
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class RegistrationWithEvent(RegistrationRead):
    event: EventListItem
